x-pack/plugins/security_solution/server/endpoint/routes/metadata/query_builders.ts

```diff
@@ -208,3 +213,74 @@ export function getESQueryHostMetadataByIDs(agentIDs: string[]) {
     index: metadataCurrentIndexPattern,
   };
 }
+
+export async function buildUnitedIndexQuery(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  request: KibanaRequest<any, any, any>,
+  endpointAppContext: EndpointAppContext,
+  ignoredAgentIds: string[] | undefined,
+  endpointPolicyIds: string[] | undefined
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Promise<Record<string, any>> {
+  const pagingProperties = await getPagingProperties(request, endpointAppContext);
+  const statusesToFilter = request?.body?.filters?.host_status ?? [];
+  const statusesKuery = buildStatusesKuery(statusesToFilter);
+
+  const filterIgnoredAgents =
+    ignoredAgentIds && ignoredAgentIds.length > 0
+      ? {
+          must_not: { terms: { 'agent.id': ignoredAgentIds } },
```

Comment:
do we need to support HostDetails... like the legacy query here? if this is being written by endpoint I'm guessing it's possible they can run an older version which still uses HostDetails?